src/handlers/coins.ts

```diff
@@ -0,0 +1,57 @@
+import { FastifyRequest } from "fastify";
+
+import { Logger } from "../logger";
+import { AuthenticateRequestParameters, AuthenticationService } from "../services/authentication";
+import { UserService } from "../services/user";
+
+const logger: Logger = new Logger({
+    name: "OsuCoinsHandler",
+});
+
+interface UserCoinsParameters extends AuthenticateRequestParameters {
+    action: "check" | "use" | "recharge" | "earn";
+    // count
+    c: number;
```

Comment:
fastify doesn't know how to deal with the types so you must always use `string` - it won't cast. you'll need to `parseInt` because of that too

Reply:
![image](https://github.com/user-attachments/assets/6d9273ee-d2a6-4b7d-982a-f09e19a9b790)
wdym